### ISSUE TITLE
Change `no-unneeded-ternary` rule to comparing tokens

### ISSUE THREAD

**Issue by ota-meshi** [eslint/eslint#13486]
**What rule do you want to change?**

- `no-unneeded-ternary`

**Does this change cause the rule to produce more or fewer warnings?**

Produce more warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Makes the process of detecting duplicates compare by token, like `no-dupe-else-if`.

**Please provide some example code that this change will affect:**

```js
var foo = a ? a : bar; // Already reported.
var foo = a.b ? a.b : bar; // Will be reported.
var foo = a.b() ? a.b() : bar; // Will be reported.
```

**What does the rule currently do for this code?**

The comparison is for only one Identifier.

**What will the rule do after it's changed?**

Duplicates can also be reported for `a.b` and `a.b()`.  
However, in this case, if there are side effects, I think it is necessary to prevent autofix.

**Are you willing to submit a pull request to implement this change?**

If there is consensus, I will submit a pull request.

---

refs https://github.com/eslint/eslint/issues/13466#issuecomment-654867869

**Comment by anikethsaha** [comment]
Thanks for the issue. 👍 

`no-unneeded-ternary`  is a `stylistic` rule and according to [new policies](https://eslint.org/blog/2020/05/changes-to-rules-policies) no enhancement is being accepted for `stylistic` rule.

**Comment by mdjermanovic** [comment]
I think this would certainly need to be behind an option, which would be indeed a new option in a stylistic rule, and that falls under the new policy.

But, I believe the current behavior might be an oversight. From #3260 it looks like the intention was to check more than just identifiers, but then it got merged without a clear resolution.

Thoughts from the team?
